### PR TITLE
CASMINST-3657 - CSI basecamp and dnsmasq gateway cleanup

### PR DESCRIPTION
**Pull request by trad511**
#### Summary and Scope

- Fixes CASMINST-3657

##### Issue Type

- Bugfix Pull Request https://github.com/Cray-HPE/cray-site-init/pull/101

For early boot with the PIT, both basecamp (BSS) and DNSMasq (DHCP/DNS) used the m001/PIT as the system gateway.  While this works for MTL, NMN and HMN for initial PXE boot because the network is singular (technically the same Layer 2), there is both inconsistent handling of the gateway between HMN and other networks, and really inconsistent overrides for CMN and CAN networks.

This change allows the networkBuiilder, including the supernet hack, to create all the network data and then assigns the following consistently for BSS/cloud-init data and DNSMasq configuration:

1. DNS is assigned to the bootstrap_dhcp PIT address for the network.
2. NTP is assigned to the bootstrap_dhcp PIT address for the network.
3. The network gateway is assigned from the bootstrap_dhcp gateway address (which follows either the ToR gateway on the switches or the Spine gateway on the switches via the supernet hack).

For DNSMasq this results in the following changes (Surtur data):
* No change for statics.conf
* No change for MTL.conf, NMN.conf
* No change for CAN.conf or CMN.conf
* Changes in HMN.conf:

Original:
```
dhcp-option=interface:bond0.hmn0,option:router,10.254.1.20
```
New:
```
dhcp-option=interface:bond0.hmn0,option:router,10.254.0.1
```

**Additional Changes:**
* Fixed the mis-use of a pointer to Subnets in`dnsmasq.go`. This changed gateway values in the main network structures which erroneously changed cloud-init data generated by `basecamp.go`.
* Removed unused `networks` directory which should have been removed when network yaml file generation was removed.
* Cleaned up the use of strings.

Original NCN metadata from data.json:
```
    "14:02:ec:d9:7a:60": {
        "meta-data": {
            "local-hostname": "ncn-m002",
...snip...
            "ipam": {
                "can": {
                    "gateway": "10.103.0.129",
                    "ip": "10.103.0.141/26",
                    "parent_device": "bond0",
                    "vlanid": 6
                },
                "chn": {
                    "gateway": "10.103.0.193",
                    "ip": "10.103.0.205/26",
                    "parent_device": "bond0",
                    "vlanid": 5
                },
                "cmn": {
                    "gateway": "10.103.0.1",
                    "ip": "10.103.0.27/25",
                    "parent_device": "bond0",
                    "vlanid": 7
                },
                "hmn": {
                    "gateway": "10.254.1.22",
                    "ip": "10.254.1.20/17",
                    "parent_device": "bond0",
                    "vlanid": 4
                },
                "mtl": {
                    "gateway": "10.1.1.11",
                    "ip": "10.1.1.10/16",
                    "parent_device": "bond0",
                    "vlanid": 0
                },
                "nmn": {
                    "gateway": "10.252.0.1",
                    "ip": "10.252.1.12/17",
                    "parent_device": "bond0",
                    "vlanid": 2
                }
            }
        },
```
New NCN metadata from data.json (note hmn and mtl gateway changes):
```
    "14:02:ec:d9:7a:60": {
        "meta-data": {
...snip...
            "ipam": {
                "can": {
                    "gateway": "10.103.0.129",
                    "ip": "10.103.0.141/26",
                    "parent_device": "bond0",
                    "vlanid": 6
                },
                "chn": {
                    "gateway": "10.103.0.193",
                    "ip": "10.103.0.205/26",
                    "parent_device": "bond0",
                    "vlanid": 5
                },
                "cmn": {
                    "gateway": "10.103.0.1",
                    "ip": "10.103.0.11/25",
                    "parent_device": "bond0",
                    "vlanid": 7
                },
                "hmn": {
                    "gateway": "10.254.0.1",
                    "ip": "10.254.1.20/17",
                    "parent_device": "bond0",
                    "vlanid": 4
                },
                "mtl": {
                    "gateway": "10.1.0.1",
                    "ip": "10.1.1.10/16",
                    "parent_device": "bond0",
                    "vlanid": 0
                },
                "nmn": {
                    "gateway": "10.252.0.1",
                    "ip": "10.252.1.12/17",
                    "parent_device": "bond0",
                    "vlanid": 2
                }
            }
        },
```

#### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I tested this on internal system (x) (if yes, please include results or a description of the test)
 
#### Idempotency
 
Should improve idempotency since data is not being overwritten.
 
#### Risks and Mitigations
 
* Should reduce risk by reducing side effects.
* BSS/cloud-init data on existing systems is not remediated by this change.

### CODE REVIEW

**Review by heemstra — APPROVED** [review]

**Review by spillerc-hpe — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Comment by trad511** [comment]
I'd like to wait for review from @johren-hpe to make sure there aren't any unintended consequences.

**Review by rustydb — APPROVED** [review]

**Review by johren-hpe — APPROVED** [review]

**Review by heemstra — APPROVED** [review]

**Pull request merged**
